lib/fs.js

```diff
@@ -1437,14 +1437,16 @@ function readdirSyncRecursive(basePath, options) {
     );
     handleErrorFromBinding(ctx);
 
-    for (let i = 0; i < readdirResult.length; i++) {
-      if (withFileTypes) {
+    if (withFileTypes) {
+      for (let i = 0; i < readdirResult[0].length; i++) {
```

Comment:
can you assign this length to a variable with a meaningful name, and add a comment of why we are not accessing the `readdirResult` but `readdirResult[0]`? It's hard to keep track why are we are getting the `[0]`